app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt

```diff
@@ -257,6 +261,39 @@ class NavigationOverlayFragment : Fragment() {
         activity!!.moveTaskToBack(true)
     }
 
+    private fun observeAccountState(): Disposable {
+        val fxaRepo = serviceLocator.fxaRepo
+
+        return fxaRepo.accountState.subscribe {
+            when (it) {
+                FxaRepo.AccountState.AUTHENTICATED_WITH_PROFILE -> {
+                    fxaRepo.accountManager.accountProfile()?.let { profile ->
+                        if (profile.avatar != null) {
+                            profile.avatar?.url?.let { url ->
+                                PicassoWrapper.client
+                                    .load(url)
+                                    .into(fxaButton)
+                            }
+                        } else {
+                            fxaButton.setImageResource(R.drawable.ic_avatar_authenticated_no_picture)
+                            fxaButton.contentDescription = resources.getString(R.string.fxa_navigation_item_signed_in2)
+                        }
+                    }
+                }
+                FxaRepo.AccountState.AUTHENTICATED_NO_PROFILE -> {
+                    fxaButton.setImageResource(R.drawable.ic_avatar_authenticated_no_picture)
+                    fxaButton.contentDescription = resources.getString(R.string.fxa_navigation_item_signed_in2)
+                }
+                FxaRepo.AccountState.NEEDS_REAUTHENTICATION,
+                FxaRepo.AccountState.NOT_AUTHENTICATED -> {
+                    fxaButton.setImageResource(R.drawable.ic_fxa_login)
```

Comment:
nit: we can de-duplicate the assignment logic by putting the "all resources are local" case into a helper function:
```kotlin
function setIconAndContentDescription() {
    fxaButton.setImageResource(...)
    fxaButton.contentDescription = ...
}

NOT_AUTHENTICATED -> {
    setIconAndContentDescription(...)
}
```

This may seem pedantic but I think it helps prevent one branch from assigning one asset but not the other.

Reply:
I like it